File: api/internal/domain/cms/cms_test.go
package cms_test

import (
	"testing"

	"github.com/vernonedu/entrepreneurship-api/internal/domain/cms"
)

func TestStatusConstants(t *testing.T) {
	tests := []struct{ name, val, want string }{
		{"draft", cms.StatusDraft, "draft"},
		{"published", cms.StatusPublished, "published"},
		{"archived", cms.StatusArchived, "archived"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if tt.val != tt.want {
				t.Errorf("got %q, want %q", tt.val, tt.want)
			}
		})
	}
}

func TestCategoryConstants(t *testing.T) {
	cats := []string{cms.CategoryTipsKarir, cms.CategoryInfoKursus, cms.CategoryBerita, cms.CategoryEvent}
	for _, c := range cats {
		if c == "" {
			t.Errorf("category constant must not be empty")
		}
	}
}

func TestFaqCategoryConstants(t *testing.T) {
	cats := []string{
		cms.FaqCategoryUmum,
		cms.FaqCategoryPendaftaran,
		cms.FaqCategoryPembayaran,
		cms.FaqCategorySertifikat,
		cms.FaqCategoryProgramKarir,
	}
	for _, c := range cats {
		if c == "" {
			t.Errorf("faq category constant must not be empty")
		}
	}
}

func TestCmsPage_BasicFields(t *testing.T) {
	p := &cms.CmsPage{Slug: "home", Title: "Home Page"}
	if p.Slug != "home" || p.Title != "Home Page" {
		t.Error("CmsPage fields not set correctly")
	}
}

func TestCmsArticle_BasicFields(t *testing.T) {
	a := &cms.CmsArticle{Slug: "test-article", Title: "Test", Status: cms.StatusDraft}
	if a.Status != cms.StatusDraft {
		t.Errorf("expected %q, got %q", cms.StatusDraft, a.Status)
	}
}

func TestCmsFaq_BasicFields(t *testing.T) {
	f := &cms.CmsFaq{Question: "?", Answer: "A", PageSlugs: []string{"home"}}
	if len(f.PageSlugs) != 1 || f.PageSlugs[0] != "home" {
		t.Error("CmsFaq PageSlugs not set correctly")
	}
}

func TestCmsTestimonial_BasicFields(t *testing.T) {
	tst := &cms.CmsTestimonial{StudentName: "Alice", Quote: "Great!", Rating: 5}
	if tst.Rating != 5 {
		t.Errorf("expected rating 5, got %d", tst.Rating)
	}
	if tst.StudentName != "Alice" {
		t.Errorf("expected student name Alice, got %s", tst.StudentName)
	}
}

func TestCmsMedia_BasicFields(t *testing.T) {
	m := &cms.CmsMedia{FileName: "photo.jpg", FileType: "image/jpeg", FileSize: 1024}
	if m.FileName != "photo.jpg" {
		t.Errorf("expected FileName photo.jpg, got %s", m.FileName)
	}
	if m.FileSize != 1024 {
		t.Errorf("expected FileSize 1024, got %d", m.FileSize)
	}
}

File: frontend/src/schemas/__tests__/cms.test.ts
import { describe, it, expect } from 'vitest'
import { updateCmsPageSchema } from '../cmspage'
import { createCmsArticleSchema, updateCmsArticleSchema } from '../cmsarticle'
import { createCmsFaqSchema } from '../cmsfaq'
import { createCmsTestimonialSchema } from '../cmstestimonial'
import { uploadCmsMediaSchema } from '../cmsmedia'

describe('updateCmsPageSchema', () => {
  it('accepts valid title', () => {
    const r = updateCmsPageSchema.safeParse({ title: 'Home' })
    expect(r.success).toBe(true)
  })

  it('rejects empty title', () => {
    const r = updateCmsPageSchema.safeParse({ title: '' })
    expect(r.success).toBe(false)
  })
})

describe('createCmsArticleSchema', () => {
  const VALID = { title: 'How to', category: 'tips', content: 'Body', status: 'draft' }

  it('accepts valid input', () => {
    expect(createCmsArticleSchema.safeParse(VALID).success).toBe(true)
  })

  it('rejects empty content', () => {
    expect(createCmsArticleSchema.safeParse({ ...VALID, content: '' }).success).toBe(false)
  })

  it('rejects invalid status', () => {
    expect(createCmsArticleSchema.safeParse({ ...VALID, status: 'foo' }).success).toBe(false)
  })

  it('update accepts slug', () => {
    expect(updateCmsArticleSchema.safeParse({ ...VALID, slug: 'how-to' }).success).toBe(true)
  })
})

describe('createCmsFaqSchema', () => {
  it('accepts valid', () => {
    const r = createCmsFaqSchema.safeParse({ question: 'Q?', answer: 'A.' })
    expect(r.success).toBe(true)
  })

  it('rejects empty question', () => {
    expect(createCmsFaqSchema.safeParse({ question: '', answer: 'A' }).success).toBe(false)
  })

  it('defaults page_slugs to []', () => {
    const r = createCmsFaqSchema.safeParse({ question: 'Q?', answer: 'A' })
    if (r.success) expect(r.data.page_slugs).toEqual([])
  })
})

describe('createCmsTestimonialSchema', () => {
  const VALID = { student_name: 'John', quote: 'Great', rating: 5 }

  it('accepts valid', () => {
    expect(createCmsTestimonialSchema.safeParse(VALID).success).toBe(true)
  })

  it('rejects rating > 5', () => {
    expect(createCmsTestimonialSchema.safeParse({ ...VALID, rating: 6 }).success).toBe(false)
  })

  it('rejects rating < 1', () => {
    expect(createCmsTestimonialSchema.safeParse({ ...VALID, rating: 0 }).success).toBe(false)
  })
})

describe('uploadCmsMediaSchema', () => {
  it('accepts valid', () => {
    const r = uploadCmsMediaSchema.safeParse({
      url: 'https://x.com/a.png',
      file_name: 'a.png',
      file_type: 'image/png',
      file_size: 1024,
    })
    expect(r.success).toBe(true)
  })

  it('rejects empty url', () => {
    expect(
      uploadCmsMediaSchema.safeParse({
        url: '',
        file_name: 'a.png',
        file_type: 'image/png',
      }).success,
    ).toBe(false)
  })
})
